### ISSUE TITLE
Mypy does not recognize pkg_resources-style namespace packages

### ISSUE THREAD

**Issue by Morikko** [python/mypy#14052]
**Bug Report**

Mypy is not able to understand [pkg_resources-style namespace packages](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/#pkg-resources-style-namespace-packages). Although, this is not the recommended namespace mechanism, it is still supported in Python for legacy compatibility. Especially as [the documentation does not recommend to mix](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/#creating-a-namespace-package) this method with [PEP420 namespace](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/#native-namespace-packages).

**To Reproduce**

A project with such structure:

```
./root/
    __init__.py # Namespace
    core/
        action.py
        __init__.py
./extensions/
    root/
        __init__.py  # Namespace
        extra/
            __init__.py
            extra_action.py
```
where inside `__init__.py` as namespace, you have:

```python
__import__("pkg_resources").declare_namespace(__name__)
```

**Expected Behavior**

```bash
$> mypy .             
Success: no issues found in 6 source files
```

**Actual Behavior**

```bash
$> mypy .             
root/__init__.py: error: Duplicate module named "root" (also at "./extensions/root/__init__.py")
root/__init__.py: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules for more info
root/__init__.py: note: Common resolutions include: a) using `--exclude` to avoid checking one of them, b) adding `__init__.py` somewhere, c) using `--explicit-package-bases` or adjusting MYPYPATH
Found 1 error in 1 file (errors prevented further checking)
```

Note: If all the `__init__.py` as namespace are removed such that it falls back to PEP420 namespace then mypy is happy:

```
$> mypy .             
Success: no issues found in 4 source files
```

**Your Environment**

- Mypy version used: `990`
- Mypy command-line flags: `mypy . `
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.9.1
